internal/commands/server/k8s_e2e_test.go

```diff
@@ -1018,48 +1228,47 @@ func tcpRouteStatusCheck(ctx context.Context, resources *resources.Resources, ga
 	}
 }
 
-func routeRefErrors(conditions []meta.Condition) bool {
-	for _, condition := range conditions {
-		if condition.Type == "ResolvedRefs" &&
-			condition.Status == "False" &&
-			condition.Reason == "Errors" {
-			return true
+func createConditionsCheck(expected []meta.Condition) func([]meta.Condition) bool {
+	return func(actual []meta.Condition) bool {
+		for _, eCondition := range expected {
+			matched := false
+			for _, aCondition := range actual {
+				if aCondition.Type == eCondition.Type &&
+					aCondition.Status == eCondition.Status &&
+					// Match if expected condition doesn't define an expected reason
+					(aCondition.Reason == eCondition.Reason || eCondition.Reason == "") {
+					matched = true
+					break
+				}
+			}
+
+			if !matched {
+				return false
+			}
 		}
+		return true
 	}
-	return false
 }
 
 func conditionAccepted(conditions []meta.Condition) bool {
-	for _, condition := range conditions {
-		if condition.Type == "Accepted" ||
-			condition.Status == "True" {
```

Comment:
Refactor uncovered that this should've been `&&`, so this check was erroneously passing on any condition with `Status: "True"`, and was being called in a few places where `conditionReady` should have been called instead.